### DISCUSSION TITLE
Issues with registerGradient()

### DISCUSSION THREAD

**Discussion by flokleiser** [hvianna/audioMotion-analyzer#89] [Q&A]
I would love to add a custom gradient that is grayscale to my project, but when i try to call "audioMotion.registerGradient('test' {...options etc}) on my projects page and then run it, i get an AudioMotionError about "Unknown gradient: 'test'". Is there anything i have to look out for to make audioMotion recognize a gradient, or are there any examples of people implementing their own gradients? Of course i could just put the whole audioMotion-anayzer.js into my src/ folder, edit that, and then import it from there, but i would like to set the gradient in my script. I would appreciate any help :)

**Comment by hvianna** [comment]
It looks like you're trying to set the gradient *before* registering it.

Try this:

```
import AudioMotionAnalyzer from 'https://cdn.skypack.dev/audiomotion-analyzer?min';

const audioMotion = new AudioMotionAnalyzer(
  document.getElementById('container'),
  {
    source: document.getElementById('audio'),
    mode: 3,
  }
);

audioMotion.registerGradient( 'grayscale', {
  colorStops: [ '#fff', '#000' ]
});

audioMotion.gradient = 'grayscale';
```

**Comment by flokleiser** [comment]
thanks for the reply, that worked. i assumed i could apply the custom gradient in the same place where i could set the other options like modes etc., but just calling "audiomotion.gradient" separately worked perfectly👍
